src/geotiff.js

```diff
@@ -610,25 +610,26 @@ class MultiGeoTIFF extends GeoTIFFBase {
    * @returns {GeoTIFFImage} the image at the given index
    */
   async getImage(index = 0) {
-    if (!this.fileDirectoriesPerFile) {
-      if (!this.fileDirectoriesPerFileParsing) {
-        this.fileDirectoriesPerFileParsing = this.parseFileDirectoriesPerFile();
+    await this.getImageCount();
+    await this.parseFileDirectoriesPerFile();
```

Comment:
I think this could be improved in the future. The idea is to delay the parsing of the IFDs as much as possible, especially when multiple files are involved, as this process can be really slow since the IFDs could be scattered all over, resulting in as many reads as there are IFDs.
For now, I would leave it as is.